### PR TITLE
Amqp open handshake

### PR DESCRIPTION
**Pull request by ilyaanisimov-kaazing**

### CODE REVIEW

File: service/amqp/pom.xml [hunk 1]
```diff
@@ -1,129 +1,184 @@
...
-        <groupId>org.kaazing</groupId>
-        <artifactId>gateway</artifactId>
-        <version>develop-SNAPSHOT</version>
-        <relativePath>../../pom.xml</relativePath>
-    </parent>
-    <modelVersion>4.0.0</modelVersion>
-    <artifactId>gateway.service.amqp</artifactId>
-    <name>Kaazing WebSocket Gateway - AMQP Service</name>
-    <description>WebSocket Gateway AMQP Service</description>
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (cmebarrow) [inline service/amqp/pom.xml]: Indentation has changed. Did you add tab characters or replace tab characters with spaces? Our standard is never to include tab characters in source files (some existing files may not always respect this but it is the objective).

File: service/amqp/src/main/scripts/org/kaazing/gateway/service/amqp/amqp091/ws.accept.rpt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#
```
Comment (cmebarrow) [inline service/amqp/src/main/scripts/org/kaazing/gateway/service/amqp/amqp091/ws.accept.rpt]: Any reason why you can't just use ws.accept.rpt script from your new k3po spec project?

File: service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            // @formatter:off
+            GatewayConfiguration configuration = new GatewayConfigurationBuilder()
+                    .service()
+                    .accept("wsn://localhost:8333/amqp")
+                    .connect("tcp://localhost:8334")
+                    .type("amqp.proxy")
+                    .property("service.domain","localhost")
+                    .property("encryption.key.alias", "session")
+                    .realmName("demo")
```
Comment (cmebarrow) [inline service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java]: We don't need the security stuff here (realm and security element). That's for HTTP level authentication. AMQP level identity (in open) is different.

File: service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+    
+    
+    @Test
+    @ScriptProperty({ "connectLocation \"http://localhost:8333/amqp\"", "acceptLocation \"tcp://localhost:8334\"" })
+    @Specification({ "specification/amqp/ws/ws.connect", "specification/amqp/ws/open/identity/request", "specification/amqp/ws/close/request",  
+        "specification/amqp/tcp/open/tcp.accept", "specification/amqp/tcp/open/identity/response", "specification/amqp/tcp/close/response"})
+    public void openAndCloseHandshake() throws Exception {
```
Comment (cmebarrow) [inline service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java]: I would expect that you would have a method here for each method in the k3po AmqpIndenityTcpIT (or at least two methods, one for openWithIdentityAndClose and one for openWithoutIdentityAndClose)

File: service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            k3po.finish();
+    }
+    
+    public static class BasicLoginModule implements LoginModule {
```
Comment (cmebarrow) [inline service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java]: You do not need this here

File: transport/http/pom.xml [hunk 6]
```diff
@@ -163,6 +163,11 @@
             <groupId>org.kaazing</groupId>
```
Comment (cmebarrow) [inline transport/http/pom.xml]: Why is this file included in your PR?

File: transport/http/src/test/java/org/kaazing/gateway/transport/http/acceptor/specification/rfc7231/RequestMethodsIT.java [hunk 7]
```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
```
Comment (cmebarrow) [inline transport/http/src/test/java/org/kaazing/gateway/transport/http/acceptor/specification/rfc7231/RequestMethodsIT.java]: Why is this file included in your PR?

**Comment by cmebarrow** [comment]
@ilyaanisimov-kaazing Done reviewing

**Comment by ilyaanisimov-kaazing** [comment]
@cmebarrow please review fixes

**Comment by cmebarrow** [comment]
@ilyaanisimov-kaazing Please answer all my comments before asking for re-review, that will streamline the process (e.g. I need answers about the two files that shouldn't be in this PR)

**Comment by ilyaanisimov-kaazing** [comment]
Created a replacement PR #476
